### DISCUSSION TITLE
LDAP add issue

### DISCUSSION THREAD

**Discussion by pio-gra** [CiscoDevNet/iPSK-Manager#55] [General]
Hi, I'm having an issue when adding LDAP server, I'm filling in all the fields, clicking "create" and nothing happens. The "add" window just disappears and the server is not created. Is there anything I should check or do to make it work?

**Comment by ciesinsn** [comment]
Is this still a issue?  Do you see any errors in the apache logs?

**Reply by pio-gra** [reply]
Not sure which logs should I look into. The issue is still there, I have even deployed the portal from a scratch recently.

**Reply by ciesinsn** [reply]
If you followed the instructions completely the file should be named admin-error.log and usually is in /var/log/apache2. 

Are you able to share what you are entering into the fields on the LDAP screen?

**Reply by pio-gra** [reply]
Actually I am still using the old iPSK manager version for production, and now I'd like to migrate to the new one. I am basically entering the same credentials and configuration that works fine with the old deployment. After clicking "Create" it just disappears and nothing happens. Here is what I see in the log file:

user@host:~$ sudo cat /var/log/apache2/admin-error.log
[Thu Nov 28 00:00:00.323746 2024] [ssl:warn] [pid 1052] AH01906: 127.0.0.1:443:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
[Thu Nov 28 00:00:00.323790 2024] [ssl:warn] [pid 1052] AH01909: 127.0.0.1:443:0 server certificate does NOT include an ID which matches the server name
[Thu Nov 28 10:10:39.586558 2024] [php7:notice] [pid 264938] [client 10.239.30.86:64377] PHP Notice:  Trying to access array offset on value of type bool in /var/www/iPSK-Manager/adminportal/index.php on line 103
[Thu Nov 28 10:10:42.222324 2024] [php7:notice] [pid 264938] [client 10.239.30.86:64377] PHP Notice:  Trying to access array offset on value of type bool in /var/www/iPSK-Manager/adminportal/login.php on line 64, referer: https://url_address/
[Thu Nov 28 10:10:42.312958 2024] [php7:notice] [pid 264938] [client 10.239.30.86:64377] PHP Notice:  Undefined index: log-purge-interval in /var/www/iPSK-Manager/adminportal/login.php on line 114, referer: https://url_address/
[Thu Nov 28 15:31:12.100754 2024] [php7:notice] [pid 280933] [client 10.46.128.69:57908] PHP Notice:  Trying to access array offset on value of type bool in /var/www/iPSK-Manager/adminportal/index.php on line 103
[Thu Nov 28 15:31:39.069799 2024] [php7:notice] [pid 264941] [client 10.46.128.69:57934] PHP Notice:  Trying to access array offset on value of type bool in /var/www/iPSK-Manager/adminportal/login.php on line 64, referer: https://url_address/
[Thu Nov 28 15:31:39.165060 2024] [php7:notice] [pid 264941] [client 10.46.128.69:57934] PHP Notice:  Undefined index: log-purge-interval in /var/www/iPSK-Manager/adminportal/login.php on line 114, referer: https://url_address/

**Reply by pio-gra** [reply]
The interesting thing is that I basically copied the LDAP setup from the "old" database to the new one (SQL), the entry was of course there, I was able to do the successful connection test but it wasn't working for the user authentication etc.

**Reply by ciesinsn** [reply]
Noting in that log output would be for the ldap pages.  If you tail -f the log does anything appear after you press create? 

When you say move from old database to new one did you copy the values from the GUI, or, did you copy using SQL from one to the other?  

Are you using any special characters in any of the fields you are entering data, if so, can you share which field and what special character?  

I'm not able to reproduce so trying to figure out what is different.

**Reply by pio-gra** [reply]
I just copied the SQL directly, whole LDAP row. There is nothing in the "live log" when I'm trying to configure LDAP, there are no special characters - just a basic url's and DN's.

**Reply by ciesinsn** [reply]
The ldap table was changed in the recent versions of iPSK Manager where a column was added.  Did you use a copy method that including dropping and rebuilding the table?  Was filing in this column part of your copy process?   This could explain some issues.

**Reply by pio-gra** [reply]
I noticed that the new structure is different, and modified the SQL query to copy only the "right" values, after that I filled in the rest manually (I think it's only about the LDAP type).

**Reply by ciesinsn** [reply]
Yes, that is the only field and it should be set to 0 for anyone using LDAP in the older versions.  Have you had the developer tools open when submitting a new LDAP server to see if anything is logged in the console log?

**Reply by pio-gra** [reply]
This is what I see in the log
![image](https://github.com/user-attachments/assets/60fd2d1c-bfdc-4b60-b199-fed48c0f8208)

**Reply by ciesinsn** [reply]
Can you switch to the console tab that is where errors will show.  What you attached is the network tab.

**Reply by pio-gra** [reply]
Noticed something like that
![image](https://github.com/user-attachments/assets/00fee78a-0056-42cc-ba09-52bd30ee123b)

**Reply by ciesinsn** [reply]
That is a interesting message as I don't see it, but, the modal is the display screen of entering the new ldap server, editing a ldap server, and viewing.  When exactly does that pop up?   Have you cleared your browser cache after installing the new version?  What browser and version is this?  Have you tried a different browser?

**Reply by pio-gra** [reply]
I am using Brave browser (1.73.91) but tried with Firefox and Edge as well. The behavior is exactly the same - after clicking "Create", it just disappears.

**Reply by ciesinsn** [reply]
My recommendation is you install a fresh installation of iPSK Manager with a empty database and attempt to add the LDAP server in.  If it works then it's most likely something in how you migrated data.  If it doesn't then you'll need to send me what you are inputing as I can't reproduce a issue saving new LDAP servers.   If you didn't want to spend a lot of time setting up a new iPSK Manager deployment you can use the Docker container that's referenced in the README what's new section.

**Reply by pio-gra** [reply]
Sorry for the lack of information from my side, so it ended up with redeployment of Ubuntu and in the new environment it worked fine.

**Reply by ciesinsn** [reply]
Thanks for following up with the information.
